### PR TITLE
Add error strings to reduce how many `GameActions::Result`s have `STR_NONE, STR_NONE`

### LINKED ISSUE #15782: Review and change game action result messages
A lot of game actions use no messages or some sort of substitute, all of them should be reviewed after #15637 is merged.

An example:
https://github.com/OpenRCT2/OpenRCT2/blob/a4ed1cb6ced5e036e0a29f5d562a3326f31cd796/src/openrct2/actions/BalloonPressAction.cpp#L51-L55

Right now it has no title or error message, this should ideally say something like "Invalid entity index" which could be STR_INVALID_PARAMETER for the title and STR_INVALID_ENTITY_INDEX for the message.

This is just an example, ideally one extracts all lines with inappropriate messages to make a nice table.

### LINKED ISSUE #21159: Some game actions return `STR_NONE` in the `GameActions::Result`, which makes an empty error message box and prevents plugins from showing a custom error
### Operating System

Windows 10 64-bit

### OpenRCT2 build

OpenRCT2, v0.4.7 (0e8d46e) provided by GitHub

### Base game

RollerCoaster Tycoon 2, RollerCoaster Tycoon Classic

### Area(s) with this issue?

This is a development issue

### Describe the issue

For example in [StaffFireAction](https://github.com/OpenRCT2/OpenRCT2/blob/296e8cdb1685f01aa9183f1cec17d5e712de57e4/src/openrct2/actions/StaffFireAction.cpp#L64), here's what happens if you try to fire a staff ID which doesn't exist:

```C++
LOG_ERROR("Invalid spriteId. spriteId = %u", _spriteId);
return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
```

The game automatically shows a error box for the unsuccessful action, but since the title and message are both empty the box is empty. Then calling `ui.showError` in a plugin does not work!

In StringIds.h I didn't find a generic "invalid" message. Should we add one??

### Steps to reproduce

(1) Add this plugin:

```javascript
registerPlugin({
    name: 'Error message example',
    version: '1.0',
    authors: ['pfroud'],
    type: 'remote',
    licence: "MIT",
    targetApiVersion: 81,
    main: function () {
        ui.registerMenuItem("Fire handyman with ID 0", function () {
            const args = {
                id: 0
            };
            const callback = function (result) {
                console.log("Result from trying to fire handyman with ID 0:")
                console.log("Error number:  " + result.error);
                console.log("Error title:   " + result.errorTitle);
                console.log("Error message: " + result.errorMessage);
                if (result.error !== 0) {
                    ui.showError("my error title", "my error message");
                }
            };
            context.executeAction("stafffire", args, callback);
        });
    }
});
```

(2) Load a scenario with no staff. Click the map icon in the top bar and click the plugin's item.

The GUI only shows an empty red error message box:

empty red error box

The log window shows:

```
ERROR[D:\a\OpenRCT2\OpenRCT2\src\openrct2\actions\StaffFireAction.cpp:48 (StaffFireAction::Query)]: Invalid spriteId. spriteId = 0
'Result from trying to fire handyman with ID 0:'
'Error number:  1'
'Error title:   '
'Error message: '
```

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by pfroud**
Reduces how many `GameAction`s are created with no error title or message, for example

```C++
GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
```

Closes #15782 and Closes #21159.

Adds these strings:
```C++
STR_ERR_INVALID_PARAMETER
STR_ERR_VALUE_OUT_OF_RANGE
STR_ERR_GHOST_ELEMENT_NOT_FOUND
STR_ERR_BALLOON_NOT_FOUND
STR_ERR_STAFF_NOT_FOUND
STR_ERR_RIDE_NOT_FOUND
STR_ERR_RIDE_OBJECT_ENTRY_NOT_FOUND
STR_ERR_PLAYER_NOT_FOUND
STR_ERR_ENTRANCE_ELEMENT_NOT_FOUND
STR_ERR_SURFACE_ELEMENT_NOT_FOUND
STR_ERR_TILE_ELEMENT_NOT_FOUND
STR_ERR_TRACK_ELEMENT_NOT_FOUND
STR_ERR_TRACK_BLOCK_NOT_FOUND
STR_ERR_PATH_ELEMENT_NOT_FOUND
STR_ERR_WALL_ELEMENT_NOT_FOUND
STR_ERR_BANNER_ELEMENT_NOT_FOUND
```

### CODE REVIEW

File: src/openrct2/actions/RideEntranceExitRemoveAction.cpp [hunk 1]
```diff
@@ -94,14 +94,15 @@ GameActions::Result RideEntranceExitRemoveAction::Query() const
...
     if (entranceElement != nullptr && (GetFlags() & GAME_COMMAND_FLAG_GHOST) && !(entranceElement->IsGhost()))
     {
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
+        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_GHOST_ELEMENT_NOT_FOUND, STR_NONE);
     }
     else if (entranceElement == nullptr)
     {
         LOG_WARNING(
-            "Track Element not found. x = %d, y = %d, ride = %u, station = %u", _loc.x, _loc.y, _rideIndex.ToUnderlying(),
+            "Entrance not found. x = %d, y = %d, ride = %u, station = %u", _loc.x, _loc.y, _rideIndex.ToUnderlying(),
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideEntranceExitRemoveAction.cpp:102]:
```suggestion
            "Entrance element not found. x = %d, y = %d, ride = %u, station = %u", _loc.x, _loc.y, _rideIndex.ToUnderlying(),
```

File: src/openrct2/actions/RideEntranceExitRemoveAction.cpp [hunk 2]
```diff
@@ -130,14 +131,15 @@ GameActions::Result RideEntranceExitRemoveAction::Execute() const
...
     if (entranceElement != nullptr && isGhost && !(entranceElement->IsGhost()))
     {
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
+        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_GHOST_ELEMENT_NOT_FOUND, STR_NONE);
     }
     else if (entranceElement == nullptr)
     {
         LOG_WARNING(
-            "Track Element not found. x = %d, y = %d, ride = %u, station = %d", _loc.x, _loc.y, _rideIndex.ToUnderlying(),
+            "Entrance not found. x = %d, y = %d, ride = %u, station = %d", _loc.x, _loc.y, _rideIndex.ToUnderlying(),
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideEntranceExitRemoveAction.cpp:139]: idem

Comment (Gymnasiast) [inline src/openrct2/actions/RideEntranceExitRemoveAction.cpp:139]:
```suggestion
            "Entrance element not found. x = %d, y = %d, ride = %u, station = %d", _loc.x, _loc.y, _rideIndex.ToUnderlying(),
```

**Review by Gymnasiast — APPROVED** [review]

File: src/openrct2/actions/RideEntranceExitRemoveAction.cpp [hunk 3]
```diff
@@ -94,14 +94,15 @@ GameActions::Result RideEntranceExitRemoveAction::Query() const
     // If we are trying to remove a ghost and the element we found is real, return an error, but don't log a warning
     if (entranceElement != nullptr && (GetFlags() & GAME_COMMAND_FLAG_GHOST) && !(entranceElement->IsGhost()))
     {
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
+        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_GHOST_ELEMENT_NOT_FOUND, STR_NONE);
```
Comment (tupaschoal) [inline src/openrct2/actions/RideEntranceExitRemoveAction.cpp:97]: Why is this not `INVALID_PARAMTER` + specific error, like the others?
```suggestion
        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_INVALID_PARAMETER, STR_ERR_GHOST_ELEMENT_NOT_FOUND);
```

File: src/openrct2/actions/RideEntranceExitRemoveAction.cpp [hunk 4]
```diff
@@ -130,14 +131,15 @@ GameActions::Result RideEntranceExitRemoveAction::Execute() const
     // If we are trying to remove a ghost and the element we found is real, return an error, but don't log a warning
     if (entranceElement != nullptr && isGhost && !(entranceElement->IsGhost()))
     {
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
+        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_GHOST_ELEMENT_NOT_FOUND, STR_NONE);
```
Comment (tupaschoal) [inline src/openrct2/actions/RideEntranceExitRemoveAction.cpp:134]: Same question as above

File: src/openrct2/actions/StaffFireAction.cpp [hunk 5]
```diff
@@ -39,14 +39,14 @@ GameActions::Result StaffFireAction::Query() const
     if (_spriteId.ToUnderlying() >= MAX_ENTITIES || _spriteId.IsNull())
     {
         LOG_ERROR("Invalid spriteId. spriteId = %u", _spriteId);
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
+        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_INVALID_PARAMETER, STR_ERR_STAFF_NOT_FOUND);
```
Comment (tupaschoal) [inline src/openrct2/actions/StaffFireAction.cpp:42]: Isn't this one out of range?
```suggestion
        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_INVALID_PARAMETER, STR_ERR_VALUE_OUT_OF_RANGE);
```

File: src/openrct2/actions/StaffSetCostumeAction.cpp [hunk 6]
```diff
@@ -64,21 +64,22 @@ GameActions::Result StaffSetCostumeAction::Query() const
 {
     if (_spriteIndex.ToUnderlying() >= MAX_ENTITIES || _spriteIndex.IsNull())
     {
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
+        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_INVALID_PARAMETER, STR_ERR_STAFF_NOT_FOUND);
```
Comment (tupaschoal) [inline src/openrct2/actions/StaffSetCostumeAction.cpp:67]: Same question as above

File: src/openrct2/actions/StaffSetOrdersAction.cpp [hunk 7]
```diff
@@ -45,7 +45,7 @@ GameActions::Result StaffSetOrdersAction::Query() const
 {
     if (_spriteIndex.ToUnderlying() >= MAX_ENTITIES || _spriteIndex.IsNull())
     {
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
+        return GameActions::Result(GameActions::Status::InvalidParameters, STR_ERR_INVALID_PARAMETER, STR_ERR_STAFF_NOT_FOUND);
```
Comment (tupaschoal) [inline src/openrct2/actions/StaffSetOrdersAction.cpp:48]: Same question

File: src/openrct2/actions/TrackSetBrakeSpeedAction.cpp [hunk 8]
```diff
@@ -64,7 +64,8 @@ GameActions::Result TrackSetBrakeSpeedAction::QueryExecute(bool isExecuting) con
     if (tileElement == nullptr)
     {
         LOG_WARNING("Invalid game command for setting brakes speed. x = %d, y = %d", _loc.x, _loc.y);
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
+        return GameActions::Result(
+            GameActions::Status::InvalidParameters, STR_ERR_INVALID_PARAMETER, STR_ERR_VALUE_OUT_OF_RANGE);
```
Comment (tupaschoal) [inline src/openrct2/actions/TrackSetBrakeSpeedAction.cpp:68]:
```suggestion
        return GameActions::Result(
            GameActions::Status::InvalidParameters, STR_ERR_INVALID_PARAMETER, STR_ERR_TILE_ELEMENT_NOT_FOUND);
```

File: src/openrct2/actions/RideEntranceExitRemoveAction.cpp (hunk 1 above)
Comment (pfroud) [reply src/openrct2/actions/RideEntranceExitRemoveAction.cpp]: I did see several more `LOG_WARNING` and `LOG_ERROR` messages which need to be updated, I will submit another PR when this one is finished

**Comment by Gymnasiast** [comment]
@pfroud If you made the changes a line note requests, you should mark them as resolved.

File: src/openrct2/actions/StaffSetCostumeAction.cpp (hunk 6 above)
Comment (pfroud) [reply src/openrct2/actions/StaffSetCostumeAction.cpp]: Fixed!

File: src/openrct2/actions/StaffFireAction.cpp (hunk 5 above)
Comment (pfroud) [reply src/openrct2/actions/StaffFireAction.cpp]: You are right, that is out of range. The "staff not found" message should only be used after we try to get an entity from the sprite. I pushed a fix

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**